plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/JdkLdapClient.java

```diff
@@ -106,11 +111,45 @@ public boolean isGroupMember(String searchBase, String groupSearch, String conte
         }
     }
 
+    @Override
+    public Set<String> lookupUserGroups(String searchBase, String searchFilter, String contextUserDistinguishedName, String contextPassword)
+            throws NamingException
+    {
+        try (CloseableContext context = createUserDirContext(contextUserDistinguishedName, contextPassword); CloseableSearchResults search = searchContext(searchBase, searchFilter, context)) {
+            ImmutableSet.Builder<String> groupNames = ImmutableSet.builder();
+            if (search.hasMore()) {
+                Attributes attributes = search.next().getAttributes();
+                Attribute memberOfAttribute = attributes.get("memberof");
+                if (memberOfAttribute == null) {
+                    log.error("No memberOf attribute found... The ldap group provider requires the memberOf overlay to be enabled.");
+                }
+                else {
+                    for (Enumeration groupDns = memberOfAttribute.getAll(); groupDns.hasMoreElements(); ) {
+                        String dn = groupDns.nextElement().toString();
+                        LdapName ln = new LdapName(dn);
+                        for (Rdn rdn : ln.getRdns()) {
+                            if (rdn.getType().equalsIgnoreCase("cn")) {
+                                groupNames.add(rdn.getValue().toString());
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+            search.close();
```

Comment:
Nitpick: This probably doesn't need to be closed explicitly, since it's managed by the try-with-resources.

Reply:
Updated